src/webapp/pages/settings/SettingsPage.tsx

```diff
@@ -0,0 +1,204 @@
+import i18n from "@dhis2/d2-i18n";
+import _ from "lodash";
+import { TableAction, RowConfig, ObjectsTable, TableState, useSnackbar } from "@eyeseetea/d2-ui-components";
+import { Button } from "@dhis2/ui";
+import React, { useCallback, useState, useMemo, useEffect } from "react";
+import { PageHeader } from "../../components/page-header/PageHeader";
+import { useGoBack } from "../../hooks/useGoBack";
+import { useAppContext } from "../../contexts/app-context";
+import { ListMetadataResponse, ListOptions } from "../../../domain/repositories/MetadataRepository";
+import { displayName, MetadataItem, MetadataModel } from "../../../domain/entities/MetadataItem";
+import Dropdown, { DropdownOption } from "../../components/dropdown/Dropdown";
+import RemoveCircleOutlineIcon from "@material-ui/icons/RemoveCircleOutline";
+import AddCircleOutlineIcon from "@material-ui/icons/AddCircleOutline";
+
+export const SettingsPage: React.FC = () => {
+    const { compositionRoot } = useAppContext();
+    const snackbar = useSnackbar();
+    const goBack = useGoBack();
+    const [builder, updateBuilder] = useState<Builder>(defaultBuilder);
+
+    const [isLoading, setIsLoading] = useState<boolean>(false);
+    const [response, setResponse] = useState<ListMetadataResponse>(initialResponse);
+    const [listOptions, setListOptions] = useState<ListOptions>(initialState);
+
+    const columns = useMemo(
+        () => [
+            { name: "name", text: i18n.t("Name"), sortable: true },
+            { name: "id", text: i18n.t("ID"), sortable: true },
+            {
+                name: "model",
+                text: i18n.t("Metadata Type"),
+                sortable: false,
+                getValue: (row: MetadataItem) => compositionRoot.metadata.getModelName(String(row.model)),
+            },
+            { name: "publicAccess", text: i18n.t("Public Access"), sortable: true },
```

Comment:
Remember that when you merge this with development and also with https://github.com/EyeSeeTea/sharing-settings-app-dev/pull/12 changes, you'll need to update this too